Source/WebGPU/WebGPU/Texture.h

```diff
@@ -162,6 +162,16 @@ class Texture : public WGPUTextureImpl, public RefCounted<Texture>, public CanMa
     mutable WeakHashSet<CommandEncoder> m_commandEncoders;
     id<MTLSharedEvent> m_sharedEvent { nil };
     uint64_t m_sharedEventSignalValue { 0 };
-};
+} SWIFT_SHARED_REFERENCE(retainTexture, releaseTexture);
 
 } // namespace WebGPU
+
+inline void retainTexture(WebGPU::Texture* obj)
+{
+    WTF::retainRefCounted(obj);
```

Comment:
WTF::retainRefCounted still needs to be cleaned up. It can just call object->ref(). No need for RELEASE_ASSERT, static_assert, or static_cast.